Port Driver Station Dashboard Articles from Screensteps (#662)
wpilibsuite/frc-docs: source/docs/software/driverstation/manually-setting-the-driver-station-to-start-dashboard.rst

Manually Setting the Driver Station to Start Dashboard
======================================================

.. note:: If WPILib is not installed to the default location (such as when files are copied to a PC manually), the dashboard of choice may not launch properly. To have the DS start a dashboard when it starts up, you have to manually modify the settings for the default dashboard.

> .. warning:: This is not needed for most installations, try using the appropriate :ref:`Dashboard Type setting <docs/software/driverstation/driver-station:Setup Tab>` for your language first. Also, to use a dashboard without installing WPILib, Java 11 must be installed first.

Reviewer: ```suggestion
.. warning:: This is not needed for most installations, try using the appropriate :ref:`Dashboard Type setting <docs/software/driverstation/driver-station:Setup Tab>` for your language first. Also to use SmartDashboard or Shuffleboard, Java 11 must be installed first.
```

Set Driver Station to Default
-----------------------------

.. image:: images/manually-setting-the-driver-station-to-start-smartdashboard/set-ds-to-default.png

Open the Driver Station software, click on the Setup tab and set the Dashboard setting to Default. **Then close the Driver Station!**

Locate Dashboard JAR file
-------------------------

> Find the location of the dashboard jar file (SmartDashboard or Shuffleboard). It should be installed to C:\Users\USERNAME\wpilib\tools where USERNAME is the username of the user logged into the computer.

Reviewer: Need to escape the path

Reviewer 2: Or better, replace it with an inline formatting block ``C:\Users\Public\Documents\FRC``

Open DS Data Storage file
-------------------------

.. image:: images/manually-setting-the-driver-station-to-start-smartdashboard/open-ds-data-storage-file.png

> Browse to "C:\Users\Public\Documents\FRC" and double click on "FRC DS Data Storage" to open it.

Reviewer: Need to escape path

Reviewer 2: Or better, replace it with an inline formatting block ``C:\Users\Public\Documents\FRC``

DashboardCmdLine
----------------

.. image:: images/manually-setting-the-driver-station-to-start-smartdashboard/dashboard-cmd-line.png

Locate the line beginning with **DasboardCmdLine**.

For SmartDashboard
^^^^^^^^^^^^^^^^^^

> Replace the string after = with **"java -jar "C:\\Users\\USERNAME\\wpilib\\tools\\SmartDashboard.jar""** where Username is the username determined above. Save the file.

Reviewer: Should add a note somewhere that java 11 is required for SmartDashboard and Shuffleboard

Reviewer 2: It'll use the built in wpilib java 11

Reviewer: If wpilib was installed with the installer, this article isn't necessary as the tools would launch automatically, This is specifically for someone who's trying to set up a driver station without installing all of wpilib.

Also, as wpilib doesn't put the wpilib java on the path, just running java -jar wouldn't run the wpilib jdk, you would have to point to the path of the wpilib jdk.

Author: I'll add a note that to use a dashboard without WPILib, Java 11 is required. Does that sound good?

Reviewer: Even if WPILib is installed, just calling out java -jre would run the system java, not the wpilib java installation, so it's still necessary to have java 11 installed as the system java.

Reviewer 2: You could add a note to run ``frcvars2020``. I would encourage this instead.

Reviewer: As the wpilib java doesn't get security updates (except yearly), I don't recommend setting it as the system java. Thus you'd have to make a batch file that runs frcvars and then the dashboard. But that still assumes that wpilib is installed, in which case this article isn't necessary. I think the current wording (that incorporated my suggested change) is good.

Reviewer 2: I don't like the idea of supporting shuffleboard/smartdashboard outside of the installer. It only leads to problems.
